db.go

```diff
@@ -750,6 +750,16 @@ func (db *DB) Snapshot(dir string, withHead bool) error {
 // Querier returns a new querier over the data partition for the given time range.
 // A goroutine must not handle more than one open Querier.
 func (db *DB) Querier(mint, maxt int64) (Querier, error) {
+	// Substract the retention from the current time and convert it to milliseconds so we can compare it with mint, maxt.
+	retentionCutoff := time.Now().Add(-time.Duration(db.opts.RetentionDuration)*time.Millisecond).UnixNano() / int64(time.Millisecond)
```

Comment:
With the `UnixNano()` we are leaving the domain of the specific encoding of `time.Duration`, so it'd be better to divide by `1e6` instead of `time.Millisecond` (whose int value depends on the `time.Duration` encoding implementation details).